Realm/RLMRealm.mm

```diff
@@ -705,6 +705,37 @@ + (NSError *)migrateRealm:(RLMRealmConfiguration *)configuration {
     }
 }
 
+- (void)dispatchAsync:(dispatch_queue_t)queue handingOver:(NSArray<RLMObject *> *)objectsToHandOver
+        withBlock:(void(^)(RLMRealm *, NSArray<RLMObject *> *))block {
+    std::vector<realm::Row> rowsToHandOver;
+    rowsToHandOver.reserve(objectsToHandOver.count);
+    NSMutableArray<NSString *> *classNames = [NSMutableArray arrayWithCapacity:objectsToHandOver.count];
```

Comment:
I couldn't find a better way to store schema information than by name. We cannot store the `RLMObjectSchema` object since it is associated with a particular `RLMRealm`. Is there a way to store an index instead? I don't think that would be safe with runtime schema changes though. Is the current implementation compatible with runtime schema changes/ @tgoyne